mitodl/mitxonline
# feat: sync is_self_paced with edX

#### Pre-Flight checklist


#### What are the relevant tickets?
#1155 
The counterpart of this PR is https://github.com/mitodl/edx-api-client/pull/93

#### What's this PR do?
- Syncs the `is_self_paced` property of course run with/from edX.

#### How should this be manually tested?
- This can be reviewed together with https://github.com/mitodl/edx-api-client/pull/93
- Create 2 courses on edX and associate them with MITxOnline, e.g. adding course run with courseware_id in from edX.
- How to change the pace in edX? The pace-setting option in edX is mentioned in the ticket see https://github.com/mitodl/mitxonline/issues/1155#issuecomment-1283017903, Note that in edX the pace can only be changed if a course hasn't started yet, So if you see the pace-setting disabled, set the course start date in future.
- Also note that only those courses are synced which have started, which means if you've set the start date in the future in the above step, you'll have to set it back to the past so that the course starts syncing again

- To test this, You can manually run the task in the web shell:
```
from courses.tasks import sync_courseruns_data

sync_courseruns_data()
```
This will sync the course details like before, like the title, and dates but this time it will sync pacing too.

#### Where should the reviewer start?
- Create courses in edX and link them to MITxOnline
- For reference of edX Course details API, can be seen at https://courses.edx.org/api-docs/#/courses/courses_v1_courses_read
